src/main/java/io/github/moulberry/notenoughupdates/overlays/BazaarSearchOverlay.java

```diff
@@ -0,0 +1,572 @@
+		JsonObject bazaarInfo2 = NotEnoughUpdates.INSTANCE.manager.auctionManager.getBazaarInfo(o2);
+
+		boolean auc1Invalid = bazaarInfo1 == null || !bazaarInfo1.has("curr_sell");
+		boolean auc2Invalid = bazaarInfo2 == null || !bazaarInfo2.has("curr_sell");
+
+		if (auc1Invalid && auc2Invalid) return o1.compareTo(o2);
+		if (auc1Invalid) return -1;
+		if (auc2Invalid) return 1;
+
+		int sales1 = bazaarInfo1.get("curr_sell").getAsInt();
+		int sales2 = bazaarInfo2.get("curr_sell").getAsInt();
+
+		if (sales1 == sales2) return o1.compareTo(o2);
+		if (sales1 > sales2) return -1;
+		return 1;
+	};
+
+	public static boolean shouldReplace() {
+		if (!NotEnoughUpdates.INSTANCE.hasSkyblockScoreboard()) return false;
+		if (!NotEnoughUpdates.INSTANCE.config.bazaarTweaks.enableSearchOverlay) return false;
+
+		if (!(Minecraft.getMinecraft().currentScreen instanceof GuiEditSign)) {
+			if (!NotEnoughUpdates.INSTANCE.config.bazaarTweaks.keepPreviousSearch) searchString = "";
+			return false;
+		}
+
+		String lastContainer = Utils.getLastOpenChestName();
+		if (!lastContainer.startsWith("Bazaar ➜ ")) return false;
+
+		TileEntitySign tes = ((AccessorGuiEditSign) Minecraft.getMinecraft().currentScreen).getTileSign();
+
+		if (tes == null) return false;
+		if (tes.getPos().getY() != 0) return false;
+		if (!tes.signText[2].getUnformattedText().equals("^^^^^^^^^^^^^^^")) return false;
+		return tes.signText[3].getUnformattedText().equals("Enter query");
+	}
+
+	public static void render() {
+		ScaledResolution scaledResolution = new ScaledResolution(Minecraft.getMinecraft());
+		int width = scaledResolution.getScaledWidth();
+		int height = scaledResolution.getScaledHeight();
+		int mouseX = Mouse.getX() * width / Minecraft.getMinecraft().displayWidth;
+		int mouseY = height - Mouse.getY() * height / Minecraft.getMinecraft().displayHeight - 1;
+
+		Utils.drawGradientRect(0, 0, width, height, -1072689136, -804253680);
+
+		int h = NotEnoughUpdates.INSTANCE.config.bazaarTweaks.showPastSearches ? 219 : 145;
+
+		int topY = height / 4;
+		if (scaledResolution.getScaleFactor() >= 4) {
+			topY = height / 2 - h / 2 + 5;
+		}
+
+		Minecraft.getMinecraft().getTextureManager().bindTexture(SEARCH_OVERLAY_TEXTURE);
+		GlStateManager.color(1, 1, 1, 1);
+		Utils.drawTexturedRect(width / 2 - 100, topY - 1, 203, h, 0, 203 / 512f, 0, h / 256f, GL11.GL_NEAREST);
+
+		Minecraft.getMinecraft().fontRendererObj.drawString("Enter Query:", width / 2 - 100, topY - 10, 0xdddddd, true);
+
+		textField.setFocus(true);
+		textField.setText(searchString);
+		textField.setSize(149, 20);
+		textField.setCustomBorderColour(0xffffff);
+		textField.render(width / 2 - 100 + 1, topY + 1);
+
+		if (textField.getText().trim().isEmpty()) autocompletedItems.clear();
+
+		List<String> tooltipToDisplay = null;
+
+		int num = 0;
+		synchronized (autocompletedItems) {
+			String[] autoCompletedItemsArray = autocompletedItems.toArray(new String[0]);
```

Comment:
This seems very redundant when it can just be replaced by an enhanced for loop and an incrementing int which would look a lot cleaner and be more performant.

```suggestion
			int i = -1;
			for (String str : autocompletedItems) {
			    i++;
			    // ...
			}
```

or you can keep a copy of the set as a list because you do this `toArray` in a few locations.

Reply:
Not my code. I copied it all from ah search overlay. There it is working fine right now, so I guess it's not necessary to change is now. You can edit both classes ah and bz search overlay later if you'd like. Or just make the code change yourself in this PR right now, I don't care. But I'm not gonna edit synchronized logics since I don't know enough about it.